buildSrc/src/test/java/org/elasticsearch/gradle/precommit/DependencyLicensesTasksIT.java

```diff
@@ -38,24 +38,27 @@ public void testDependenciesOk() {
     public void testNoDependenciesWithFolder() {
         GradleRunner runner = getRunner("no_dependencies_with_folder");
 
-        assertTaskFailed(runner.build(), ":no_dependencies_and_no_folder:dependencyLicenses");
+        assertTaskFailed(runner.buildAndFail(), ":no_dependencies_and_no_folder:dependencyLicenses");
     }
 
     public void testDependenciesMissing() {
         GradleRunner runner = getRunner("dependencies_missing");
 
-        assertTaskFailed(runner.build(), ":dependencies_missing:dependencyLicenses");
+        assertTaskFailed(runner.buildAndFail(), ":dependencies_missing:dependencyLicenses");
```

Comment:
This change is needed, but when asserting for failure we should assert for something in the output to make sure we are getting the failure we want.
I can see that from the last test run we got `Plugin with id 'elasticsearch.build' not found.` so I don't think we are getting the intended failure here yet the test would pass.

The reason for the failure is that you must use the new Gradle syntax to apply the plugin. take a look at `buildSrc/src/testKit/elasticsearch.build/build.gradle`